components/butterfly/src/rumor/dat_file.rs

```diff
@@ -322,8 +322,8 @@ impl DatFile {
               .map_err(|err| Error::DatFileIO(path.to_path_buf(), err))?;
         debug!("Header Version: {}", version[0]);
 
-        // If this has happened, it's likely that the file is corrupt
-        if version[0] > HEADER_VERSION {
+        // If this has happened, it's likely that the file was mutated in some unexpected way.
+        if version[0] == 0 || version[0] > HEADER_VERSION {
```

Comment:
The answer is almost assuredly "no"... I did a bit of spelunking, and version 2 was introduced back in version 0.27.0, which is 3 years ago. If anybody is still actively using 0.27.0, they have much bigger problems than the format of this file :joy:

However, if we're going to commit to removing support for version 1, we should do that in a dedicated manner in its own PR.

In the meantime, it may be more clarifying to create a `OLDEST_VERSION = 1` constant and reframe this conditional as
```rust
if version[0] < OLDEST_VERSION || version[0] > HEADER_VERSION
```
(Changing `HEADER_VERSION` to `CURRENT_VERSION` might be a further improvement.)

Reply:
I also thought about doing this but then realized it might be a trap we're setting for our future selves.  If we only support the _current_ version, then how can we safely upgrade? Given that the Dat file implementation is perhaps a little brittle and not subject to change very often unless we completely abandon it for something else, I think it might be best to support all known versions (currently just 1, 2) so that we can always ensure an upgrade path exists.

Reply:
I went with

```
if version[0] < OLDEST_HEADER_VERSION || version[0] > CURRENT_HEADER_VERSION {
```